linera-service/src/server.rs

```diff
@@ -395,6 +395,36 @@ enum ServerCommand {
         #[arg(long, default_value = "1000")]
         cache_size: usize,
     },
+
+    /// Replaces the configurations of the shards.
+    #[command(name = "edit-shards")]
+    EditShards {
+        /// Path to the file containing the server configuration of this Linera validator.
+        #[arg(long = "server")]
+        server_config_path: PathBuf,
+
+        /// The number N of shard configs to generate (1 <= N < 100).
+        #[arg(long)]
+        num_shards: u8,
+
+        /// The host of the validator (IP address or hostname). `%%` will be replace by
+        /// the shard number.
+        #[arg(long)]
+        host: String,
+
+        /// The port of the main endpoint.
+        #[arg(long)]
+        port: u16,
```

Comment:
I prefer [NonZeroUsize](https://doc.rust-lang.org/std/num/type.NonZeroUsize.html) for ports.

Reply:
I think [`NonZeroU16`](https://doc.rust-lang.org/std/num/type.NonZeroU16.html) is a better fit.